### DISCUSSION TITLE
Legion Slim 5 16IRH8 - completely disable 4070 on battery

### DISCUSSION THREAD

**Discussion by Miademora** [BartoszCichecki/LenovoLegionToolkit#1102] [Q&A] [answered]
Hello,

Lenovo Vantage disables the 4070 in battery mode completely so it vanishes in task manager.
With Legion Toolkit i see it switching between inactive and powered off and it remains in task manager.
Its using quite a bit more battery for me. 

Can LegionToolkit do this as well? I tried searching for a separate script to disable it completely. I cant find anything not even a bios switch.

Regards
Michel

**Comment by BartoszCichecki** [comment, marked as answer]
It's called Hybrid-Auto and it's also in LLT. If it doesn't disappear on battery, something is blocking dGPU from getting "ejected".

**Comment by Miademora** [comment]
Thanks for the answer. While investigating that Hybrid-Auto didnt seem to eject the graphics card i found the following.

There is no process blocking it. I tried killing as much as i could.
When i switch to Hybrid and back Hybrid-Auto in LTT it instantly ejects the DGPU.
Is this something that LTT controls?

Good to have a workaround but i have to remember it.
It doesnt seem available as under actions either.

**Reply by BartoszCichecki** [reply]
Hybrid-Auto isn't the most reliable thing on the planet, but it works most of the time. Ejecting dGPU after unplugging AC is initiated by EC, so if it doesn't happen, maybe a future BIOS version will fix it.

**Reply by Miademora** [reply]
good to know. thanks. can you expose it under actions maybe?

**Reply by BartoszCichecki** [reply]
It already is exposed. Check README for instructions on how to turn it on - it's an optional feature.
